src/client/hooks/useProducts.js

```diff
@@ -1,23 +1,44 @@
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 
-const fetchProducts = () => {
-  const url = '/api/products';
+const fetchProducts = (limit, offset, sortKey, sortOrder) => {
+  const url = `/api/products?limit=${limit}&offset=${offset}&sortKey=${sortKey}&sortOrder=${sortOrder}`;
   return fetch(url).then((response) => response.json());
 };
 
-export const useProducts = () => {
-  const [products, setProducts] = useState();
-  const [isLoading, setLoading] = useState(true);
+export const PRODUCT_PER_PAGE = 12;
+
+export const useProducts = ({ sortKey, sortOrder }) => {
+  const [products, setProducts] = useState([]);
+  const [totalCount, setTotalCount] = useState(0);
+  const [isLoading, setLoading] = useState(false);
   const [error, setError] = useState();
 
-  if (!products && !error) {
-    fetchProducts()
-      .then((productsResponse) => setProducts(productsResponse))
+  const loadMoreProducts = () => {
```

Comment:
We should wrap this in `useCallback` and rename it to something like `loadPage` since it will load a single page from the API.

The `useCallback` needs a dependency list that should look something like:
`[isLoading, sortKey, sortOrder, currentPage]`